### PR TITLE
Implement datetime64 as the time dtype

### PR DESCRIPTION
**Pull request by lkilcher**
@jmcvey3- I'm definitely interested in you reviewing this. I think this is a major improvement because now we're using real time objects!

Note that one of the big stumbling blocks I think you ran into was fixed in a25008e. Take a look at the details of [`datetime.fromtimestamp`](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp), [`datetime.utcfromtimestamp`](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp), and [`datetime.timestamp`](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp). The fine print there is important! What a mess!

Hurrah!

### CODE REVIEW

**Comment by lkilcher** [comment]
Also, I need you to help me fix all these build errors.

File: dolfyn/time.py [hunk 1]
```diff
@@ -8,14 +9,32 @@ def _fullyear(year):
...
 
-def epoch2date(ds_time, offset_hr=0, to_str=False):
+def epoch2dt64(ep_time, ):
+    out = np.array(ep_time).astype('datetime64[s]')  # assumes t0=1970-01-01 00:00:00
+    out = out + ((ep_time % 1) * 1e9).astype('timedelta64[ns]')
+    return out
+
+
+def dt642epoch(dt64):
+    return dt64.astype('datetime64[ns]').astype('int') / 1e9
```
Comment (jmcvey3) [inline dolfyn/time.py:19]: Was having issues on particular datafiles for some reason that are fixed with:
`epoch = dt64.astype('datetime64[ns]').astype('float') / 1e9`
`return epoch.astype('int')`
Not entirely sure why.

**Comment by jmcvey3** [comment]
@lkilcher, if you run
`dlfn.io.rdi.read_rdi('RDI_withBT.000')` does this pass?
`dlfn.read_example('RDI_withBT.000')` and does this fail?
The former passes but the latter fails for me, and I am very confused.

**Comment by jmcvey3** [comment]
Having EOF issues when running `dlfn.io.rdi.read_rdi('RDI_withBT.000')` and other datafiles.
Not sure why

File: dolfyn/time.py (hunk 1 above)
Comment (jmcvey3) [reply dolfyn/time.py:19]: Converting to 'int' chops off the nanosecond data. Looking for a better approach.

Comment (jmcvey3) [reply dolfyn/time.py:19]: Solved in my branch of "timestamps" (https://github.com/jmcvey3/dolfyn/commits/timestamps)

**Comment by lkilcher** [comment]
@jmcvey3 can we add a windows build (Python 3.9 probably) to the set of tests?

**Comment by lkilcher** [comment]
Also, should we drop Py3.7 and add Py3.10?

File: dolfyn/time.py (hunk 1 above)
Comment (lkilcher) [reply dolfyn/time.py]: @jmcvey3 is this resolved with one of your commits, or is this still outstanding?

**Pull request merged**